### PR TITLE
Implement shared heartbeat and stream_reset helpers (1.1.3)

### PR DESCRIPTION
**Pull request by leynos**
## Summary
- Implements the actual plan for task 1.1.3: shared SSE heartbeat policy and a fixed `stream_reset` helper, wiring them into the existing SSE framing primitives and docs. Includes new modules, tests, and updated ADR/docs to reflect the wire-only surface and ADR decisions. See new exec plan documentation for goals, constraints, risks, and validation gates.
- Updates repository docs to reference the new exec plan and outline the wire-only surface, without introducing runtime scheduling or lifecycle concerns.

## Changes
- docs/contents.md: Link to the new execution plan for 1.1.3.
- docs/execplans/1-1-3-shared-heartbeat-and-stream-reset-helpers.md: New detailed exec plan outlining goals, constraints, tolerances, risks, progress, and validation gates.
- src/sse/heartbeat.rs: New module implementing the default heartbeat interval, a typed HeartbeatPolicy with explicit override validation, and a canonical render_heartbeat_frame helper.
- src/sse/stream_reset.rs: New module implementing a fixed stream_reset helper with canonical event name and payload, built on top of render_event_frame.
- src/sse/mod.rs: Re-exports for the new heartbeat and stream_reset helpers and constants.
- src/sse/heartbeat.rs (new) and src/sse/stream_reset.rs (new): Added unit tests covering default behavior, override validation, deterministic frame output, and edge cases.
- src/lib.rs: Re-exports for DEFAULT_HEARTBEAT_INTERVAL, HeartbeatPolicy, HeartbeatPolicyError, STREAM_RESET_EVENT_NAME, STREAM_RESET_REPLAY_UNAVAILABLE_PAYLOAD, render_heartbeat_frame, and render_stream_reset_frame.
- docs/adr-001-shared-sse-wire-contract-for-wildside-and-corbusier.md: ADR updated to reference the explicit heartbeat policy semantics and fixed stream_reset surface.
- docs/developers-guide.md: Updated to document the new heartbeat policy and stream_reset helpers as wire-only, with module layout and test strategies.
- docs/users-guide.md: Updated usage examples for HeartbeatPolicy and render_heartbeat_frame, plus fixed stream_reset frame rendering examples.
- docs/roadmap.md: Marked 1.1.3 as completed. See ADR 001 references and exec plan.
- src/sse/frame.rs, src/sse/event_id.rs, src/sse/replay_cursor.rs, src/sse/cache_control.rs, and related modules remain the same, now complemented by the new helpers and re-exports.
- Updated tests to cover both new helpers and their deterministic outputs, including repeated rendering checks.

## Why this matters
- Establishes a clear, wire-only API surface for a default 20s heartbeat and a canonical stream_reset event, enabling downstream consumers to rely on consistent behavior without embedding scheduling or app lifecycle logic in the crate.
- Documents decisions and acceptance criteria up front to guide implementation and ensure ADR alignment (ADR 001 referenced).
- Keeps the implementation focused on transport-level concerns, preventing leakage of timers, schedulers, or lifecycle management into the crate.

## Build and validation plan (per exec plan)
- Stage A approvals and ADR clarifications before code landed (now completed).
- Stage B/C unit tests with rstest where appropriate; avoid unnecessary behavioural tests unless they add value.
- Stage D documentation updates across ADRs, users guide, developers guide, and roadmap; publish validation gates.
- Validation gates to run when implementing: `make check-fmt`, `make lint`, `make test`, `make fmt`, `make markdownlint`, and `make nixie`.

## Risks and constraints
- Surface must remain wire-only; avoid timers, schedulers, or lifecycle logic.
- Public API should be small and well-typed; avoid exposing raw strings or overly generic helpers.
- If implementation scope expands beyond a few files or ~350 lines, re-scope per plan.

## Approval
- Implementation proceeds under the updated ADR guidance and the new exec plan. The surface remains transport-focused and does not introduce scheduling or responder surfaces.

## Next steps after this update
- With 1.1.3 completed, update roadmap/ADR references as needed and consider any follow-up cleanup or minor refinements based on upstream feedback.

## Task reference
- Task: https://www.devboxer.com/task/d977747f-592a-49a9-bd53-b3b3ee12e98a

## Summary by Sourcery
Document the execution plan for introducing shared SSE heartbeat and stream_reset helpers and link it into the docs index.

Documentation:
- Add a new exec plan detailing goals, constraints, risks, stages, and validation for the shared SSE heartbeat and stream_reset helpers.
- Update the documentation contents index to reference the new 1.1.3 execution plan for heartbeat and stream_reset helpers.

## Summary by Sourcery

Add shared SSE heartbeat and stream_reset helpers and wire them into the public SSE API and documentation.

New Features:
- Introduce a typed HeartbeatPolicy with a default 20-second interval, validation for explicit overrides, and a canonical heartbeat frame renderer.
- Add a fixed stream_reset helper that emits the standard replay-unavailable control event with shared name and payload constants.

Documentation:
- Document the heartbeat policy and stream_reset helpers in the users and developers guides, including their wire-only scope and usage examples.
- Extend ADR 001 to capture heartbeat override semantics and the fixed stream_reset surface, and add a new execution plan for task 1.1.3.
- Update the docs index and roadmap to reference and mark completion of the 1.1.3 heartbeat and stream_reset work.

Tests:
- Add unit tests for HeartbeatPolicy defaults, override validation, and heartbeat frame determinism.
- Add unit tests for stream_reset frame rendering, determinism, and alignment between constants and rendered output.

### CODE REVIEW

**Comment by leynos** [comment]
@coderabbitai Have the following now been resolved?

Annul any requirements that violate the en-GB-oxendict spelling (-ize / -yse / -our) conventions (for example a request to replace "normalize" with "normalise" or "artefact" with "artifact"), or where the requirement unnecessarily increases cyclomatic complexity.

~~~
Please address the comments from this code review:

## Overall Comments
- In `heartbeat.rs` and `stream_reset.rs`, consider importing `render_comment_frame`/`render_event_frame` directly from the sibling modules (e.g. `crate::sse::frame::render_event_frame`) instead of via the `crate::sse` re-exports to keep intra-module dependencies clearer and avoid relying on re-export ordering.
- You might want to add `#[must_use]` to `HeartbeatPolicy::new`, `render_heartbeat_frame`, and `render_stream_reset_frame` since ignoring their results is almost certainly a bug and this would provide better compile-time feedback to callers.

## Individual Comments

### Comment 1
<location path="docs/users-guide.md" line_range="169" />
<code_context>

+### Shared heartbeat helper
+
+Use `HeartbeatPolicy` when your endpoint needs the shared heartbeat cadence in
+typed form:
+
</code_context>
<issue_to_address>
**issue (review_instructions):** This sentence uses the second-person pronoun "your", which the documentation guidelines prohibit.

To comply with the style guide, rephrase to avoid second-person. For example: "Use `HeartbeatPolicy` when an endpoint needs the shared heartbeat cadence in typed form:"

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Avoid 2nd person or 1st person pronouns ("I", "you", "we").

</details>
</issue_to_address>

### Comment 2
<location path="docs/users-guide.md" line_range="196" />
<code_context>
+assert_eq!(frame, ":\n\n");
+```
+
+This crate still does not schedule heartbeats for you. Applications remain
+responsible for timer ownership, background tasks, and stream lifecycle.
+
</code_context>
<issue_to_address>
**issue (review_instructions):** The phrase "for you" uses a second-person pronoun, which is disallowed by the documentation instructions.

Consider rephrasing to remove the second person, for example: "This crate still does not schedule heartbeats. Applications remain" or "This crate still does not schedule heartbeats on behalf of applications. Applications remain".

<details>
<summary>Review instructions:</summary>

**Path patterns:** `**/*.md`

**Instructions:**
Avoid 2nd person or 1st person pronouns ("I", "you", "we").

</details>
</issue_to_address>
~~~

File: src/sse/heartbeat.rs [hunk 1]
```diff
@@ -0,0 +1,165 @@
+//! Shared heartbeat policy and frame helpers for SSE streams.
+
+use std::time::Duration;
+
+use thiserror::Error;
+
+use crate::sse::{SseFrameError, frame::render_comment_frame};
+
```
[the remark this answers is not in the document]
Comment (leynos) [reply src/sse/heartbeat.rs:8]: @coderabbitai feedback from the engineer:

> No fix was needed for that comment.
> 
> In the current code, [src/sse/heartbeat.rs](https://github.com/leynos/actix-v2a/blob/implement-shared-heartbeat-reset-mlgxaa/src/sse/heartbeat.rs%3A107 "Open src/sse/heartbeat.rs:107 on GitHub (leynos/actix-v2a@implement-shared-heartbeat-reset-mlgxaa)") does not build the heartbeat wire format locally. `render_heartbeat_frame()` already delegates directly to the canonical frame helper via `render_comment_frame("")`, imported from `crate::sse::frame` at [src/sse/heartbeat.rs](https://github.com/leynos/actix-v2a/blob/implement-shared-heartbeat-reset-mlgxaa/src/sse/heartbeat.rs%3A7 "Open src/sse/heartbeat.rs:7 on GitHub (leynos/actix-v2a@implement-shared-heartbeat-reset-mlgxaa)"). The `Result<String, SseFrameError>` is propagated unchanged.
> 
> So the finding appears stale relative to the current implementation, and I left the code as-is.
